proposed/local-sdk-global-json.md

```diff
@@ -0,0 +1,112 @@
+    }
+}
+```
+
+In this configuration the host resolver would find a compatible SDK, if it exists in `.dotnet` or a machine wide location. The host resolver will consider the `additionalPaths` in the order they are defined and will stop at the first match. If none of the locations have a matching SDK then
+ will fall back to the existing SDK resolution strategy: `%PATH%` followed by machine wide installations.
+
+This lookup will stop on the first match which means it won't necessarily find the best match. Consider a scenario with a global.json that has:
+
+```json
+{
+  "sdk": {
+    "additionalPaths": [ ".dotnet" ],
+    "version": "7.0.200",
+    "rollForward": "latestFeature"
+  }
+}
+```
+
+In a scenario where the `.dotnet` directory had 7.0.200 SDK but there was a machine wide install of 7.0.300 SDK, the host resolver would pick 7.0.200 out of `.dotnet`. That location is considered first, it has a matching SDK and hence discovery stops there.
+
+The values in the `additionalPaths` property can be a relative or absolute path. When a relative path is used it will be resolved relative to the location of global.json. These values also support the following substitutions:
+
+- `"$user$"`: this matches the user-local installation point of .NET SDK for the current operating system: `%LocalAppData%\Microsoft\dotnet` on Windows and `$HOME/.dotnet` on Linux/macOS.
+- `"$machine$"`: this matches the machine installation point of .NET for the current operating system.
+- `%VARIABLE%/$VARIABLE`: environment variables will be substituted. Either the Windows or Unix format can be used here and will be normalized for the operating system the host resolver executes on.
+
+This design requires us to only change the host resolver. That means other tooling like Visual Studio, VS Code, MSBuild, etc ... would transparently benefit from this change. Repositories could update global.json to have `additionalPaths` support `.dotnet` and Visual Studio would automatically find it without any design changes.
+
+## Considerations
+### Installation Points
+One item to keep in mind when considering this area is the .NET SDK can be installed in many locations. The most common are:
+
+- Machine wide
+- User wide: `%LocalAppData%\Microsoft\dotnet` on Windows and `$HOME/.dotnet` on Linux/macOS.
+- Repo: `.dotnet`
+
+Our installation tooling tends to avoid redundant installations. For example, if restoring a repository that requires 7.0.400, the tooling will not install it locally if 7.0.400 is installed machine wide. It also will not necessarily delete the local `.dotnet` folder or the user wide folder. That means developers end up with .NET SDK installs in all three locations but only the machine wide install has the correct ones.
+
+As a result solutions like "just use .dotnet, if it exists" fall short. It will work in a lot of casse but will fail in more complex scenarios. To completely close the disconnect here we need to consider all the possible locations.
```

Comment:
```suggestion
As a result solutions like "just use .dotnet, if it exists" fall short. It will work in a lot of cases but will fail in more complex scenarios. To completely close the disconnect here we need to consider all the possible locations.
```